zcash_primitives/src/zip32/fingerprint.rs

```diff
@@ -0,0 +1,53 @@
+//! Seed Fingerprints according to ZIP 32
+//!
+//! Implements section `Seed Fingerprints` of Shielded Hierarchical Deterministic Wallets (ZIP 32)
+//!
+//! [Section Seed Fingerprints]: https://zips.z.cash/zip-0032#seed-fingerprints
+
+use blake2b_simd::Params as Blake2bParams;
+
+pub const ZIP32_SEED_FP_PERSONALIZATION: &[u8; 16] = b"Zcash_HD_Seed_FP";
+
+pub struct SeedFingerprint([u8; 32]);
+
+impl SeedFingerprint {
+    pub fn from_seed(seed_bytes: &[u8]) -> SeedFingerprint {
+        let mut h = Blake2bParams::new()
+            .hash_length(32)
+            .personal(ZIP32_SEED_FP_PERSONALIZATION)
+            .to_state();
+        let seed_len_bytes = seed_bytes.len().to_le_bytes();
+
+        let concatenated_input = [&seed_len_bytes, seed_bytes].concat();
+
+        h.update(&concatenated_input);
+
+        let mut seed_fp = [0u8; 32];
+        seed_fp.copy_from_slice(h.finalize().as_bytes());
+
+        SeedFingerprint(seed_fp)
```

Comment:
```suggestion
        let seed_len = seed_bytes.len();
        assert!(32 <= seed_len && seed_len <= 252);

        SeedFingerprint(Blake2bParams::new()
            .hash_length(32)
            .personal(ZIP32_SEED_FP_PERSONALIZATION)
            .to_state()
            .update(&[u8::try_from(seed_len).unwrap()])
            .update(seed_bytes)
            .finalize()
            .as_bytes())
```

Reply:
Instead of `assert`...`unwrap` here, use `u8::try_from(seed_len).ok().map( ... )`